apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx

```diff
@@ -142,6 +143,18 @@ export const WebstudioComponentDev = ({
     }
   }, [isSelected]);
 
+  useEffect(() => {
+    // 1 means root
+    if (instanceSelector.length === 1) {
+      // If by the time root is rendered,
+      // no selected instance renders and sets `selectedInstanceIsRendered` to `true`,
+      // then it's clear that selected instance will not render at all, so we set it to `false`
+      if (selectedInstanceIsRenderedStore.get() === undefined) {
+        selectedInstanceIsRenderedStore.set(false);
+      }
+    }
+  });
+
```

Comment:
I would describe cases then it happens. As I don't understand why this logic is placed on Root (Body?)
Why it should be rendered at all and why execute useEffect on every render. What if some optimisations will be applied and this rendering will never happen?

Reply:
Ah, yes. This actually relies on `useStore(selectedInstanceSelectorStore)` above in the component. We rely on re-render after `selectedInstance` change

Reply:
Added code comment